File: Sources/App/Models/Playlist.swift
//
//  Playlist.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 16/02/2025.
//

import Vapor
import Fluent

final class Playlist: Model, @unchecked Sendable {

    // MARK: - Constants
    
    static let schema = "playlists"
    
    
    
    // MARK: - Properties
    
    @ID(custom: .id, generatedBy: .database)
    var id: Int?
    
    @Field(key: "name")
    var name: String
    
    @Field(key: "description")
    var description: String?
    
    @Field(key: "image")
    var image: String?
    
    @Children(for: \.$playlist)
    var entries: [PlaylistEntry]
    
    @Timestamp(key: "created_at", on: .create)
    var createdAt: Date?
    
    @Timestamp(key: "updated_at", on: .update)
    var updatedAt: Date?
    
    
    
    // MARK: - Construction
    
    init(_ playlist: CreatePlaylistDTO) {
        name = playlist.name
        description = playlist.description
        image = nil
    }
    
    init() {}
}

File: FinchServer/Database/Playlist.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FinchServer.Database;

[Table("playlists")]
public class Playlist {
    
    [Key]
    public int Id { get; init; }
    
    [MaxLength(255)] 
    public required string Name { get; set; }
    
    [Column(TypeName = "text")]
    public string? Description { get; set; }
    
    [MaxLength(255)] 
    public string? ImageFileName { get; set; }
    
    public required ICollection<PlaylistEntry> Entries { get; set; }
    
    public DateTime CreatedAt { get; init; }
    
    public DateTime UpdatedAt { get; set; }
}